versions-common/src/main/java/org/codehaus/mojo/versions/api/ArtifactVersions.java

```diff
@@ -122,7 +121,7 @@ public boolean equals(Object o) {
 
         return new EqualsBuilder()
                 .append(getArtifact(), that.getArtifact())
-                .append(getVersions(), that.getVersions())
+                .append(getVersions(true), that.getVersions(true))
```

Comment:
default value for field `includeSnapshots` was false - why here is true?

Reply:
ArtifactVersions always contains all versions of the artifact. No matter what the value of "includeSnapshots" was before I removed it. includeSnapshots acted as an additional value to apply on top of the collected versions.

But here, we are checking if two object instances are equal. If we set this to false, an instance with and an instance without snapshots will be equal, incorrectly.